tabpfn_client/client.py

```diff
@@ -117,19 +126,25 @@ def predict(self, train_set_uid: str, x_test):
 
         return np.array(response.json()["y_pred"])
 
-    def error_raising(self, response, method_name):
+    def error_raising(self, response, method_name, only_version_check=False):
```

Comment:
Can't we simply do all checks always? Why not?

Reply:
All checks always would mean that always a RuntimeError is raised when something is wrong. It might be desirable that, e.g. when the client tries to retrieve messages from the server but there is some server error, the user still can proceed without getting the error.

Reply:
Do you think so? Are there cases like this? I think this is kinda dangerous. I rather get some arbitrary error, then my predict call just not returning anything without any reason, no? Maybe I am understanding something false, though.

Reply:
Yes, but in the example of a predict call it would raise the error because we do not set only_version_check to True. In the other cases where only_version_check is True, e.g. during registration, the program will still terminate at the moment but with a more customized exception.
I think the question here is whether it is better to raise an error anyway and if necessary chatch it in the service_wrapper function to handle it, or to ignore the error in this function on purpose because we know we will handle it later. What do you think?

Reply:
@liam-sbhoo what is your opinion on this?

Reply:
I would suggest always checking the version actually. In general, it's simpler.
And it could also make debugging easier too: in your last example, let's say the `prediction` api doesn't work. Having the check tells us that we are working on compatible version when debugging.

Reply:
We already check the version always, right? With the new only_version_check argument we can just decide if we also want to check for other errors.

Reply:
i think we kinda talked passed each other: as a general principle let us try to always communicate as much as we know about an error to the user. in the places you use `only_version_check` it is fine, only please set it to false on login (i added a comment there too). let's resolve this.